studio/components/interfaces/TableGridEditor/SidePanelEditor/SidePanelEditor.tsx

```diff
@@ -394,6 +392,54 @@ const SidePanelEditor = ({
     resolve()
   }
 
+  const onImportData = async (importContent: ImportContent) => {
+    if (!project || selectedTable === undefined) {
+      return console.error('no project or table selected')
+    }
+
+    const { file, rowCount, selectedHeaders, resolve } = importContent
+    const toastId = ui.setNotification({
+      category: 'loading',
+      message: `Adding ${rowCount.toLocaleString()} rows to ${selectedTable.name}`,
+    })
+    const { error }: any = await meta.insertRowsViaSpreadsheet(
+      file,
+      selectedTable,
+      selectedHeaders,
+      (progress: number) => {
+        ui.setNotification({
+          id: toastId,
+          progress,
+          category: 'loading',
+          message: `Adding ${rowCount.toLocaleString()} rows to ${selectedTable.name}`,
+        })
+      }
+    )
+
+    if (error) {
+      ui.setNotification({
+        error,
+        id: toastId,
+        category: 'error',
+        message: `Failed to import data: ${error.message}`,
+      })
+      resolve()
+    } else {
+      ui.setNotification({
+        id: toastId,
+        category: 'success',
+        message: `Successfully imported ${rowCount} rows of data into ${selectedTable.name}`,
+      })
+      await Promise.all([
+        queryClient.invalidateQueries(
+          sqlKeys.query(project?.ref, [selectedTable!.schema, selectedTable!.name])
+        ),
+      ])
```

Comment:
Worth swapping the order of these two? So the toast shows after everything finishes loading